scout/server/blueprints/variants/controllers.py

```diff
@@ -366,6 +378,13 @@ def parse_variant(
             variant_obj["compounds"], key=lambda compound: -compound["combined_score"]
         )
 
+        # check compound against current query rank score
+        for compound in variant_obj["compounds"]:
+            if (query_form and query_form.get("compound_rank_score") is not None) and (
+                compound.get("rank_score") < query_form.get("compound_rank_score")
```

Comment:
I think this line is causing a potential bug. Tested on stage typing a number (3, 8066?) as value of compound_rank_score:

```
2022-03-14 08:28:28,055 - [scout.server.app](http://scout.server.app/) - ERROR: Exception on /cust000/F0006358-mip7/variants [POST] [in /venv/lib/python3.8/site-packages/flask/app.py:1457]
Traceback (most recent call last):
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/worker/app/scout/server/utils.py", line 71, in decorated_function
    context = func(*args, **kwargs)
  File "/home/worker/app/scout/server/blueprints/variants/views.py", line 126, in variants
    data = controllers.variants(
  File "/home/worker/app/scout/server/blueprints/variants/controllers.py", line 114, in variants
    parse_variant(
  File "/home/worker/app/scout/server/blueprints/variants/controllers.py", line 384, in parse_variant
    compound.get("rank_score") < query_form.get("compound_rank_score")
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```

Reply:
Yeah, I didn't think about what if the cases were unranked. It should be good now!